v03_pipeline/lib/annotations/reference_dataset_collection.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+from typing import Any
+
+import hail as hl
+
+from v03_pipeline.lib.model import (
+    DatasetType,
+    Env,
+    ReferenceDatasetCollection,
+    ReferenceGenome,
+)
+from v03_pipeline.lib.paths import reference_dataset_collection_path
+
+
+def hgmd(
+    ht: hl.Table,
+    env: Env,
+    reference_genome: ReferenceGenome,
+    dataset_type: DatasetType,
+    **_: Any,
+) -> hl.Expression:
+    if (
+        ReferenceDatasetCollection.HGMD
+        not in dataset_type.annotatable_reference_dataset_collections(env)
+    ):
+        return None
+    hgmd_ht = hl.read_table(
+        reference_dataset_collection_path(
+            env,
+            reference_genome,
+            ReferenceDatasetCollection.HGMD,
+        ),
+    )
+    return hgmd_ht[ht.key].hgmd
+
+
+def gnomad_non_coding_constraint(
+    ht: hl.Table,
+    env: Env,
+    reference_genome: ReferenceGenome,
+    dataset_type: DatasetType,
+    **_: Any,
+) -> hl.Expression:
+    if (
+        ReferenceDatasetCollection.INTERVAL_REFERENCE
+        not in dataset_type.annotatable_reference_dataset_collections(env)
+    ):
+        return None
+    interval_reference_ht = hl.read_table(
```

Comment:
I feel like we only want to read the table in once and annotate everything that comes off of it, instead of reading it in for every annotation

Reply:
I've update the logic here so we read the table in outside of this function and pass the reference in as an argument.  

I think my original thinking here was that it was actually cleaner to read the table in multiple times because the table reading is actually lazy and there's no performance cost to having multiple references to the same persisted table.  Maybe that's not true?